### PR TITLE
Dynamic invoice

### PR DESCRIPTION
**Pull request by dylancom**
This will prompt the user for an amount when needed.
Can be tested at: https://webln.dev/#/api/make-invoice
Solves: #213

### CODE REVIEW

File: src/app/router/Prompt/Prompt.jsx [hunk 1]
```diff
@@ -69,11 +65,19 @@ class Prompt extends Component {
                   />
                 }
               />
+              <Route
+                path="invoice"
+                element={
+                  <MakeInvoice
+                    invoice={this.state.args.invoice}
```
Comment (bumi) [inline src/app/router/Prompt/Prompt.jsx:72]: maybe we should call this invoiceAttributes or something like that to make clear what is in there? (in this case no bolt11 payment request/invoice)

File: src/extension/background-script/actions/ln/makeInvoice.js [hunk 2]
```diff
@@ -1,14 +1,29 @@
...
   }
   PubSub.publish(`ln.makeInvoice.start`, message);
 
-  const amount = parseInt(message.args.amount);
+  let amount;
   const memo = message.args.memo;
+  if (message.args.amount) {
+    amount = parseInt(message.args.amount);
+  } else {
+    const response = await utils.openPrompt({
```
Comment (bumi) [inline src/extension/background-script/actions/ln/makeInvoice.js:16]: I think we should refactor this pattern as it had been problematic in the past (e.g. impossible to have a loading indicator) 

I think we should do that similarly to what we are doing here: https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/ConfirmPayment/index.jsx#L22

**Review by bumi — COMMENTED** [review]
what do you think?

File: src/extension/background-script/actions/ln/makeInvoice.js (hunk 2 above)
Comment (dylancom) [reply src/extension/background-script/actions/ln/makeInvoice.js:13]: Hmm the https://webln.dev/#/api/make-invoice fails if the makeInvoice is not awaiting a response. Not sure how to integrate this pattern here.

Comment (bumi) [reply src/extension/background-script/actions/ln/makeInvoice.js:13]: we have to have the response, but here we did it like this: 
prompt -> does another action call and then returns that response back

https://github.com/getAlby/lightning-browser-extension/blob/master/src/app/screens/ConfirmPayment/index.jsx#L29-L34

Comment (dylancom) [reply src/extension/background-script/actions/ln/makeInvoice.js]: Not sure why I didn't got this working yesterday, but working now :)

**Pull request merged**